packages/client/src/views/SysAdmin/Config/DynamicModal.tsx

```diff
@@ -147,45 +226,66 @@ class DynamicModalComponent extends React.Component<IFullProps, State> {
           )
         })
         .catch(() => {
-          this.setState({
-            errorOccured: true,
-            errorMessages: this.props.intl.formatMessage(
-              messages.applicationNameChangeError
-            )
-          })
+          this.setError(
+            this.props.intl.formatMessage(messages.applicationConfigChangeError)
+          )
           valueChanged(
             NOTIFICATION_TYPE.ERROR,
-            this.props.intl.formatMessage(messages.applicationNameChangeError)
+            this.props.intl.formatMessage(messages.applicationConfigChangeError)
           )
         })
-    }
-  }
-
-  async callUpdateApplicationNameMutation(applicationName: string) {
-    try {
-      this.setState({ updatingValue: true })
-      const res = await configApplicationMutations.updateApplicationName(
-        applicationName
+    } else if (
+      modalName === GeneralActionId.NID_PATTERN &&
+      value.NID_NUMBER_PATTERN
+    ) {
+      callUpdateNIDPatternMutation(
+        value.NID_NUMBER_PATTERN,
+        this.props,
+        this.setUpdatingValue,
+        this.setError
+      )
+        .then(() => {
+          valueChanged(
+            NOTIFICATION_TYPE.SUCCESS,
+            this.props.intl.formatMessage(messages.nidPatternChangeNotification)
+          )
+        })
+        .catch(() => {
+          this.setError(
+            this.props.intl.formatMessage(messages.applicationConfigChangeError)
+          )
+          valueChanged(
+            NOTIFICATION_TYPE.ERROR,
+            this.props.intl.formatMessage(messages.applicationConfigChangeError)
+          )
+        })
```

Comment:
```suggestion
      try{
          await callUpdateNIDPatternMutation(
            value.NID_NUMBER_PATTERN,
            this.props,
            this.setUpdatingValue,
            this.setError
          )
          valueChanged(
            NOTIFICATION_TYPE.SUCCESS,
            this.props.intl.formatMessage(messages.nidPatternChangeNotification)
          )
      }catch{
        this.setError(
          this.props.intl.formatMessage(messages.applicationConfigChangeError)
        )
        valueChanged(
          NOTIFICATION_TYPE.ERROR,
          this.props.intl.formatMessage(messages.applicationConfigChangeError)
        )
      }
```

Using await instead of then will look good.